Assets/NuGet/Editor/NugetHelper.cs

```diff
@@ -1291,5 +1302,113 @@ public static Texture2D DownloadImage(string url)
 
             return result;
         }
+
+        /// <summary>
+        /// Data class returned from nuget credential providers in a JSON format. As described here:
+        /// https://docs.microsoft.com/en-us/nuget/reference/extensibility/nuget-exe-credential-providers#creating-a-nugetexe-credential-provider
+        /// </summary>
+        [System.Serializable]
+        private struct CredentialProviderResponse
+        {
+            public string Username;
+            public string Password;
+        }
+
+        /// <summary>
+        /// Possible response codes returned by a Nuget credential provider as described here:
+        /// https://docs.microsoft.com/en-us/nuget/reference/extensibility/nuget-exe-credential-providers#creating-a-nugetexe-credential-provider
+        /// </summary>
+        private enum CredentialProviderExitCode
+        {
+            Success = 0,
+            ProviderNotApplicable = 1,
+            Failure = 2
+        }
+
+        /// <summary>
+        /// Helper function to aquire a token to access VSTS hosted nuget feeds by using the CredentialProvider.VSS.exe
+        /// tool. Downloading it from the VSTS instance if needed.
+        /// See here for more info on nuget Credential Providers:
+        /// https://docs.microsoft.com/en-us/nuget/reference/extensibility/nuget-exe-credential-providers
+        /// </summary>
+        /// <param name="packageHost">The hostname where the VSTS instance is hosted (such as microsoft.pkgs.visualsudio.com</param>
+        /// <returns>The password in the form of a token, or null if the password could not be aquired</returns>
+        private static string GetPasswordFromVSTSCredentialProvider(string packageHost)
+        {
+            string credentialProviderBundleFilename = "CredentialProviderBundle.zip";
+            string credentialProviderFilename = "credentialprovider.vss.exe";
+
+            // Build the list of possible locations to find the credential provider. In order it should be local app data, paths set on the
+            // environment varaible, and lastly look at the root of the pacakges save location.
+            List<string> possibleCredentialProviderPaths = new List<string>();
+            possibleCredentialProviderPaths.Add(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "Nuget", "CredentialProviders"));
+            possibleCredentialProviderPaths.AddRange(Environment.GetEnvironmentVariable("NUGET_CREDENTIALPROVIDERS_PATH")?.Split(';') ?? new string[]{});
```

Comment:
I get this error from Unity 2017.3.1f1 - the default for the project is 2017.1.1f1, but the DLL must be backwards compatible to `5.2.0f3` which is used by the packager. Curious what version you used.

```
Assets/NuGet/Editor/NugetHelper.cs(1342,122): error CS1644: Feature `null propagating operator' cannot be used because it is not part of the C# 4.0 language specification
```

Reply:
I was testing in 2017.4.1f1, the real issue though was that I was using scripting back-end .NET 4.6. Updated the code and tested with .NET 3.5 as the backend and things seem to be working there. I don't have Unity 5.2.0f3 on had to test with.